ble/Gap.h

```diff
@@ -161,6 +161,9 @@ class Gap {
 
     typedef FunctionPointerWithContext<bool> RadioNotificationEventCallback_t;
 
+    typedef FunctionPointerWithContext<const Gap *> GapShutdownCallback_t;
```

Comment:
Thanks for pointing this out. Once the callback is executed it is not removed from the callback chain. In my opinion the best approach is to add code to the `reset()` functions in every class clearing the callback chain which would effectively reset state.